src/tight.c

```diff
@@ -335,6 +336,9 @@ static enum TJPF tight_get_jpeg_pixfmt(uint32_t fourcc)
 	case DRM_FORMAT_ABGR8888:
 	case DRM_FORMAT_XBGR8888:
 		return TJPF_RGBX;
+	case DRM_FORMAT_BGR888:
+	case DRM_FORMAT_RGB888:
```

Comment:
Shouldn't `DRM_FORMAT_RGB888` yield `TJPF_BGR` and `DRM_FORMAT_BGR888` yield `TJPF_RGB`?

Reply:
I think this conflicts with vnc clients...for instance tightvnc did not like jpeg BGR. I can make the change though if you want.

Reply:
That's odd. I understood this to be the format of the input buffer. The output format for JPEG is YCbCr, not RGB, so it's strange that this would affect the client. In what way did the client "not like" the output?

I find it surprising that your reds and blues aren't swapped because these format types are of opposite endianness.

Reply:
hmm I wonder if the jpg should use the destination format, not source?

Reply:
https://rawcdn.githack.com/libjpeg-turbo/libjpeg-turbo/2.1.x/doc/html/group___turbo_j_p_e_g.html#gafbdce0112fd78fd38efae841443a9bcf

Reply:
The colours are backwards. the nvidia card/driver i have the "source format" is BGR888. but when I use TJPF_BGR, the vnc on my phone the colours are backwards. now, this is probably an issue with the vnc client.

Reply:
DRM format is little-endian, but turbojpeg uses a big-endian format. This is why you need to return `TJPF_BGR` for `DRM_FORMAT_RGB888` and vice versa.